File: app/src/main/java/com/kcrason/dynamicpagerindicator/CustomPagerIndicator.java
package com.kcrason.dynamicpagerindicator;

import android.content.Context;
import android.support.v4.view.PagerAdapter;
import android.util.AttributeSet;
import android.view.View;

import com.kcrason.dynamicpagerindicatorlibrary.BasePagerTabView;
import com.kcrason.dynamicpagerindicatorlibrary.DynamicPagerIndicator;

/**
 * @author KCrason
 * @date 2018/1/23
 */
public class CustomPagerIndicator extends DynamicPagerIndicator {

    public CustomPagerIndicator(Context context) {
        super(context);
    }

    public CustomPagerIndicator(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public CustomPagerIndicator(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    public BasePagerTabView createTabView(PagerAdapter pagerAdapter, final int position) {
        return new CustomPagerTabView(mContext);
    }
}

File: app/src/main/java/com/kcrason/dynamicpagerindicator/CustomPagerIndicator.kt
package com.kcrason.dynamicpagerindicator

import android.content.Context
import android.support.v4.view.PagerAdapter
import android.util.AttributeSet
import com.kcrason.dynamicpagerindicatorlibrary.BasePagerTabView
import com.kcrason.dynamicpagerindicatorlibrary.DynamicPagerIndicator

/**
 * @author KCrason
 * @date 2018/1/23
 */
class CustomPagerIndicator @JvmOverloads constructor(
        context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0
) : DynamicPagerIndicator(context, attrs, defStyleAttr) {
    override fun createTabView(context: Context, pagerAdapter: PagerAdapter, position: Int): BasePagerTabView {
        return CustomPagerTabView(context)
    }
}
